pkg/credentialprovider/plugin/plugin.go

```diff
@@ -135,6 +135,10 @@ func newPluginProvider(pluginBinDir string, provider kubeletconfig.CredentialPro
 			pluginBinDir: pluginBinDir,
 			args:         provider.Args,
 			envVars:      provider.Env,
+			environ:      os.Environ,
+			run: func(cmd *exec.Cmd) error {
```

Comment:
I think @liggitt's earlier comment here was to wrap `environ` into a function that returns []string, not a wrapper for `cmd.Run()`.
```
environ: func() []string {
     return os.enviorn()
}
```
And in the tests, you can stub out the `environ` field with a func that returns env vars you want for the tests

Reply:
@andrewsykim Yes `os.Environ` is stubbed and in the test cases it returns a `[]string`, however I needed to stub `exec.Run` as well to [validate](https://github.com/kubernetes/kubernetes/pull/103231/files#diff-bc8a0587b295cf65947e96a1f74138e15f41c01276d6af54d82e265a246d7ec8R843) the overlay results.

If there is a better approach then do let me know, cc: @liggitt

Reply:
> however I needed to stub exec.Run as well to validate the overlay results.

I think you can unit test a computeEnv() method that does the overlay without stubbing all the way to Run

Reply:
@liggitt Let me know if the below approach is fine

`plugin.go`
```golang
func (e *execPlugin) ExecPlugin(ctx context.Context, image string) (*credentialproviderapi.CredentialProviderResponse, error) {
....
cmd.Env = mergeEnvVars(e.environ(), e.envVars);
err = cmd.Run()
...
}

func mergeEnvVars(osEnvVars []string, configEnvVars []config.ExecEnvVar) []string {
// merge env vars and return
}
```

Reply:
@liggitt ^^

Reply:
sure

Reply:
@liggitt Done